collections/ansible_collections/cloudkit/service/roles/manage_agents/tasks/import_agents.yaml

```diff
@@ -0,0 +1,135 @@
+- name: Display node names that will be added as agents
+  ansible.builtin.debug:
+    var: manage_agents_node_names
+
+- name: Get list of node info
+  ansible.builtin.include_role:
+    name: massopencloud.esi.node
+    tasks_from: list_nodes
+  vars:
+    node_names: "{{ manage_agents_node_names }}"
+# registers: node_info_list
+
+- name: Set filtered lists of only nodes that need importing
+  ansible.builtin.include_role:
+    name: massopencloud.esi.node
+    tasks_from: get_non_agent_nodes
+# registers: filtered_node_info
+# registers: agents
+
+- name: Get InfraEnv information
+  kubernetes.core.k8s_info:
+    kind: InfraEnv
+    api_version: agent-install.openshift.io/v1beta1
+    namespace: "{{ manage_agents_namespace }}"
+    name: "{{ manage_agents_infraenv_name }}"
+  register: infraenv
+
+- name: Set discovery_url
+  ansible.builtin.set_fact:
+    discovery_url: "{{ infraenv.resources[0].status.isoDownloadURL }}"
+  when: not discovery_url|default(false)
+
+- name: Provision nodes
+  ansible.builtin.include_role:
+    name: massopencloud.esi.node
+    tasks_from: provision_nodes
+  vars:
+    provisioning_network_name: "{{ manage_agents_provisioning_network_name }}"
+    node_info_list: "{{ filtered_node_info }}"
+    wait: true
```

Comment:
I would make the image name a var rather than rely on `discovery_url` being set; it becomes much easier to reuse `provision_nodes` that way.